### DISCUSSION TITLE
NUCLEO-H743ZI2 + RAMPS

### DISCUSSION THREAD

**Discussion by AlessandroVeller** [grblHAL/core#391] [Q&A] [answered]
**Premise**
Hello, I was planning on using a NUCLEO-H743ZI2 board with a RAMPS 1.6 shield (and DRV8825 drivers) to control the 5 steppers in the MPCNC I'm building. 

**Questions**

- I saw in the [drivers' README](https://github.com/grblHAL/drivers) that the H7 can only control up to 3 axes instead of 6 and was wondering why;
- Since the drivers for the H7 are only for some boards, would I have to write my own for the RAMPS?

I'm sorry if this isn't the right place to ask this and/or for saying something stupid, I just started looking into this.
Thanks in advance

**Comment by terjeio** [comment, marked as answer]
The STM32F7xx and STMH7xx drivers supports up to 8 axes - depending on the board/board map.
You have to add a board map for the RAMPS board, the main driver code is shared so no need to write a new driver.
[Here is a map for the Arduino Due](https://github.com/grblHAL/SAM3X8E/blob/master/grblHAL_Due/src/ramps_1.6_map.h) that you may start with - but note that the port and pin designations are different. See one of the H7 maps for how they should be defined.
One important issue to be aware of is that the RAMPS board is designed for 5V signalling, the STM32 processors uses 3.3V - IIRC the board has to be slightly modified to work with 3.3V.

**Reply by AlessandroVeller** [reply]
Thanks for the pointers :)

**Comment by mikhail11111** [comment]
RAMPS 1.6 shield doesn't fit into NUCLEO-H743ZI2 board since it has two rows of pins on the short side that don't have a reciprocal on the board.

**Reply by AlessandroVeller** [reply]
Oh yeah, I completely missed that, thanks!
Do you think something like [this](https://www.pibot.com/pibot-multi-stepper-motor-driver-board-rev2-3.html) would be a good substitute?

**Reply by mikhail11111** [reply]
the CNC shield 3.0 (https://duino.ru/cnc-shield/) should fit, although it has only 4 axes.
